packages/map-balloon/src/MapBalloon.js

```diff
@@ -22,44 +24,85 @@ const MapBalloon = ({
   animated,
   borderRadius,
   pinColor,
+  pinHighlightColor,
   pinSize,
   opacity,
   balloonSize,
   captionSize,
-  imageTextSize
+  imageTextSize,
+  reference,
+  onPinClick,
+  onBalloonClick
 }) => {
+  const interactivePin = !!onPinClick;
+  const interactiveBalloon = !!onBalloonClick;
+  const pinIconClass = interactivePin ? "wg-place" : "wg-location";
   return (
     <Wrapper x={x} y={y} opacity={opacity}>
-      <Pin className="wg-location" pinColor={pinColor} pinSize={pinSize} />
-
       {(singleImage && (
         <SingleImage
           src={singleImage}
           borderRadius={borderRadius}
-          pinSize={pinSize}
+          pinColor={pinColor}
         />
-      )) ||
-        (open &&
-          imageSet && (
+      )) || (
+        <div>
+          <Pin
+            className={`${pinIconClass} ${open ? "open" : ""}`}
+            pinColor={pinColor}
+            pinHighlightColor={pinHighlightColor}
+            pinSize={pinSize}
+            interactive={interactivePin}
+            animated={animated}
+            onClick={
+              interactivePin ? event => onPinClick(reference, event) : null
```

Comment:
Any special reason to pass `null` to onClick? Just curious (:

This could be written like this as well `onClick={event => onPinClick && onPinClick(reference, event}`

Reply:
No particular reason. I first tried `onClick={interactivePin && (event => onPinClick(reference, event))}` but this resulted in some proptype warnings when `interactivePin` was `false`. So I just changed to a ternary with `null`.

Your suggestion works with no warnings. My only concern is that we're creating a new function at every render of every balloon, even it doesn't have an `onPinClick` set.

What about `onClick={onPinClick && (event => onPinClick(reference, event))}`?